test/suites/drift/suite_test.go

```diff
@@ -57,73 +69,653 @@ var _ = AfterEach(func() { env.Cleanup() })
 var _ = AfterEach(func() { env.AfterEach() })
 
 var _ = Describe("Drift", func() {
-
-	var pod *corev1.Pod
-
+	var dep *appsv1.Deployment
+	var selector labels.Selector
+	var numPods int
 	BeforeEach(func() {
-		env.ExpectSettingsOverridden(corev1.EnvVar{Name: "FEATURE_GATES", Value: "Drift=true"})
-
-		coretest.ReplaceRequirements(nodePool, karpv1.NodeSelectorRequirementWithMinValues{
-			NodeSelectorRequirement: corev1.NodeSelectorRequirement{
-				Key:      corev1.LabelInstanceTypeStable,
-				Operator: corev1.NodeSelectorOpIn,
-				Values:   []string{"Standard_DS2_v2"},
-			}})
-
-		// Add a do-not-disrupt pod so that we can check node metadata before we disrupt
-		pod = coretest.Pod(coretest.PodOptions{
-			ObjectMeta: metav1.ObjectMeta{
-				Annotations: map[string]string{
-					karpv1.DoNotDisruptAnnotationKey: "true",
+		numPods = 1
+		// Add pods with a do-not-disrupt annotation so that we can check node metadata before we disrupt
+		dep = coretest.Deployment(coretest.DeploymentOptions{
+			Replicas: int32(numPods),
+			PodOptions: coretest.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "my-app",
+					},
+					Annotations: map[string]string{
+						karpv1.DoNotDisruptAnnotationKey: "true",
+					},
 				},
+				TerminationGracePeriodSeconds: lo.ToPtr[int64](0),
 			},
-			ResourceRequirements: corev1.ResourceRequirements{Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("0.5")}},
```

Comment:
Don't want base resource requirements, or am I missing something?

Reply:
Correct, those are now specified in each test (if/as needed)

Reply:
The core test Deployment or Pods does not have any defaults, so it will end up with no resource requirements or limits. The rest of the behaviour is up to standard Kuberneter scheduling etc. - it will respect other constraints (like taints, selectors, affinity, max pods, etc.), but from resources perspective will treat it as requiring nothing (including zero CPU and memory) and can place it anywhere (even on node with no available resources).